### PR TITLE
Reworking of the jsonrpc server

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
Alternative version for https://github.com/vincenzopalazzo/lampo.rs/pull/216

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
OK testing it on a server I get the following

```
2024-05-18T17:24:29.974Z INFO lampo_jsonrpc Accepting connection: `Ok((UnixStream { fd: FileDesc(OwnedFd { fd: 4 }), local: "/home/vp/.lampo/testnet/lampod.socket" (pathname), peer: (unnamed) }, (unnamed)))`. [lampo-jsonrpc/src/lib.rs:265]
2024-05-18T17:24:29.975Z INFO jsonrpc buffer read {"method":"funds","params":{},"id":"0","jsonrpc":". [lampo-jsonrpc/src/lib.rs:186]
```

looks like the buffer is truncated

**Review by vincenzopalazzo — COMMENTED** [review]
ACK https://github.com/vincenzopalazzo/lampo.rs/pull/219/commits/ee671abbb07b505d5ca03a480fbc816877a94dd3

File: lampo-jsonrpc/src/lib.rs [hunk 1]
```diff
@@ -118,184 +117,152 @@ impl<T: Send + Sync + 'static> JSONRPCv2<T> {
...
+                    }
+                    RPCEvent::Connect => {
+                        if event.is_hangup() || event.is_error() {
+                            log::error!(target: "jsonrpc", "an error occurs: {:?}", event);
                             continue;
                         }
 
                         if event.is_invalid() {
-                            log::info!(target: "jsonrpc", "event invalid, unregister event from the tracking one");
+                            log::warn!(target: "jsonrpc", "event invalid: {:?}", event);
```
Comment (Harshit933) [inline lampo-jsonrpc/src/lib.rs:263]: This gives me the warning everytime I run `lampo-cli --network regtest getinfo`. 
```
2024-05-24T16:13:35.386Z WARN jsonrpc event invalid: Event { key: Connect, source: Source { fd: 15, events: 519, revents: 32 } }. [lampo-jsonrpc/src/lib.rs:281]
```
Should we take care of this or should we ignore this everytime.

This is the basic flow of events whenever I am querying the node. 
1. `RPCEvent::Accept`
2. `RPCEvent::Connect && event.is_readable()`
3. `RPCEvent::Connect and the event.is_invalid()` -> This throws the warning.

**Review by Harshit933 — COMMENTED** [review]
A single warning that I encountered. Otherwise all the things were good.

**Pull request merged**